classes/class-swsales-reports.php

```diff
@@ -42,141 +73,263 @@ public static function show_reports_page() { ?>
-					<p><?php esc_html_e( 'Sale Revenue', 'sitewide-sales' ); ?></p>
-				</div>
-			</div>
-			<?php
-			// Daily Revenue Chart.
-			// Build an array with each day of sale as a key to store revenue data in.
-			$date_array_all = array();
-			$period = new \DatePeriod(
-				new \DateTime( $sitewide_sale->get_start_date( 'Y-m-d' ) ),
-				new \DateInterval('P1D'),
-				new \DateTime( $sitewide_sale->get_end_date( 'Y-m-d' ) . ' + 1 day' )
-			);
-			foreach ($period as $key => $value) {
-				$date_array_all[ $value->format('Y-m-d') ] = 0.0;
+
+		// Bail if given elements aren't SWSales_Sitewide_Sale objects.
+		foreach ( $sitewide_sales as $sitewide_sale ) {
+			if ( ! is_a( $sitewide_sale, 'Sitewide_Sales\classes\SWSales_Sitewide_Sale' ) ) {
+				return;
 			}
+		}
+		?>
+		<div id="swsales_overall_sale_performance" class="swsales_reports-box">
+			<h2 class="swsales_reports-box-title"><?php esc_html_e( 'Overall Sale Performance', 'sitewide-sales' ); ?></h2>
+			<table>
+				<thead>
+					<tr>
+						<th width="25%"></th>
+						<th><?php esc_html_e( 'Banner Reach', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Landing Page Visits', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Conversions', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Sale Revenue', 'sitewide-sales' ); ?></th>
+					</tr>
+				</thead>
+				<tbody>
+				<?php
+					$daily_chart_data = [];
+					$max_sale_days = 0;
+					foreach ( $sitewide_sales as $key => $sitewide_sale ) {
+						$diff_rate = null;
+						if ( count( $sitewide_sales ) > 1 && $key === 0 ) {
+							$diff_rate = self::build_diff_rate_array( $sitewide_sale, $sitewide_sales[1]  );
+						}
+					?>
+					<tr>
+						<th>
+							<div class="swsales_reports-sale-value">
+								<a href="<?php echo esc_url( add_query_arg( array( 'post' => $sitewide_sale->get_id(), 'action' => 'edit' ), admin_url( 'post.php' ) ) ); ?>"><?php echo esc_html( $sitewide_sale->get_name() ); ?></a>
+							</div>
+							<div class="swsales_reports-sale-value-description">
+								<?php
+									printf(
+										wp_kses_post( '%s to %s', 'sitewide-sales' ),
+										esc_html( $sitewide_sale->get_start_date() ),
+										esc_html( $sitewide_sale->get_end_date() )
+									);
+								?>
+								<?php
+									$start_date = $sitewide_sale->get_start_date('Y-m-d');
+									$end_date = $sitewide_sale->get_end_date('Y-m-d');
+
+									$start_date = date_create($start_date);
+									$end_date = date_create($end_date);
+
+									// Calculate the date difference
+									$interval = $start_date->diff($end_date);
+									echo esc_html( sprintf( _n( '(%s Day)', '(%s Days)', $interval->days, 'sitewide-sales' ), number_format_i18n( $interval->days ) ) );
+								?>
+							</div>
+							<a target="_blank" href="<?php echo SWSales_Reports::build_CSV_report_link( $sitewide_sale ); ?>" class="button button-primary button-small"><?php esc_html_e( 'Export to CSV', 'sitewide-sales' ); ?></a>
```

Comment:
```suggestion
							<a target="_blank" href="<?php echo esc_url( SWSales_Reports::build_CSV_report_link( $sitewide_sale ) ); ?>" class="button button-primary button-small"><?php esc_html_e( 'Export to CSV', 'sitewide-sales' ); ?></a>
```

Reply:
Even though we esc_url in the method there, good to esc_url "as late as possible before outputting" again here to be clear to others reading the code.